src/envoy/mixer/http_filter.cc

```diff
@@ -96,7 +96,17 @@ class Config : public Logger::Loggable<Logger::Id::http> {
           __func__);
     }
 
-    http_control_ = std::make_shared<HttpControl>(mixer_server);
+    std::map<std::string, std::string> attributes;
+    if (config.hasObject("attributes")) {
+      for (const std::string& attr : config.getStringArray("attributes")) {
+        attributes[attr] = config.getString(attr);
+      }
+    } else {
+      log().warn("No attributes is specified in the config: {}", __func__);
```

Comment:
This needs to be extended to include `upstream_cluster`. @rshriram did something similar for the fault injection filter. Basically, the problem is that if you have two upstream clusters "service_a" and "service_b" in the same http connection manager (say via weighted cluster block), you need to send two different sets of attributes about "a" and "b". Maybe make "attributes" a list of JSON objects that have "upstream_cluster" and "attributes" list.

You do not have to do this in the same PR, but send a follow-up PR that allows this customization.